internal/provider/data_source_hvn_route.go

```diff
@@ -0,0 +1,104 @@
+package provider
+
+import (
+	"context"
+	"log"
+
+	sharedmodels "github.com/hashicorp/hcp-sdk-go/clients/cloud-shared/v1/models"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"github.com/hashicorp/terraform-provider-hcp/internal/clients"
+)
+
+func dataSourceHVNRoute() *schema.Resource {
+	return &schema.Resource{
+		Description: "The HVN Route data source provides information about an existing HVN route.",
+		ReadContext: dataSourceHVNRouteRead,
+		Timeouts: &schema.ResourceTimeout{
+			Default: &hvnRouteDefaultTimeout,
+		},
+		Schema: map[string]*schema.Schema{
+			// Required inputs
+			"hvn": {
+				Description: "The `self_link` of the HashiCorp Virtual Network (HVN).",
+				Type:        schema.TypeString,
+				Required:    true,
+			},
+			"destination_cidr": {
+				Description: "The destination CIDR of the HVN route",
+				Type:        schema.TypeString,
+				Required:    true,
+			},
+			// Computed outputs
+			"self_link": {
+				Description: "A unique URL identifying the HVN route.",
+				Type:        schema.TypeString,
+				Computed:    true,
+			},
+			"target": {
+				Description: "The target of the HVN route.",
+				Type:        schema.TypeMap,
+				Elem: &schema.Schema{
+					Type: schema.TypeString,
```

Comment:
@bcmdarroch do you know if we can extend this in the future, say make it more complex than just a `string: string` kind of object?

Reply:
![Screen Shot 2021-04-28 at 4 14 58 PM](https://user-images.githubusercontent.com/28313602/116483914-e34e5d80-a83c-11eb-8986-8584f27a0635.png)
it does seem like we can keep nesting this, but will let Brenna give a better answer since I'm new to writing providers

Reply:
what I meant is now we are going to have it like:
```
target = {
  self_link = "url"
}
```
what if in the future we would want to make it:
```
target = {
  self_link = "url"
  some_number = 1
  some_object = {
     foo = "bar"
  }
}
```
how would be transition from first to second? Is the second is even possible in TF?

Reply:
looked at docs_as far as i can tell_, i don't think so, the elements must be homogenous

Reply:
I don't know much about this either, but looking at an example like this, it might be possible to do? https://github.com/hashicorp/terraform-provider-aws/blob/4bd169a4786a8a35674fa7b93b4dadc17d8d9090/aws/resource_aws_eks_cluster.go#L165-L214

Reply:
thanks for the example! perhaps `type` should be `schema.TypeList` then, to allow more extensibility

Reply:
spoke to @smaant , we will make this field `target_link` instead to keep the structure flat and simple for now. in the future we may add `target` if we need to